modules/account-lib/src/coin/hbar/walletInitializationBuilder.ts

```diff
@@ -0,0 +1,101 @@
+import { BaseCoin as CoinConfig } from '@bitgo/statics/dist/src/base';
+import * as hex from '@stablelib/hex';
+import BigNumber from 'bignumber.js';
+import { proto } from '../../../resources/hbar/protobuf/hedera';
+import { BuildTransactionError } from '../baseCoin/errors';
+import { TransactionBuilder } from './transactionBuilder';
+import { Transaction } from './transaction';
+import { isValidPublicKey } from './utils';
+import { KeyPair } from './';
+
+const DEFAULT_M = 3;
+export class WalletInitializationBuilder extends TransactionBuilder {
+  private _owners: string[] = [];
+  private _txBody: proto.TransactionBody;
+  private _txBodyData: proto.CryptoCreateTransactionBody;
+  private readonly _duration: proto.Duration = new proto.Duration({ seconds: 120 });
+
+  constructor(_coinConfig: Readonly<CoinConfig>) {
+    super(_coinConfig);
+    this._txBody = new proto.TransactionBody();
+    this._txBody.transactionValidDuration = this._duration;
+    this._txBodyData = new proto.CryptoCreateTransactionBody();
+    this._txBody.cryptoCreateAccount = this._txBodyData;
+    this._txBodyData.autoRenewPeriod = new proto.Duration({ seconds: 7890000 });
+  }
+
+  /**
+   * Set one of the owners of the multisig wallet.
+   *
+   * @param {string} address The public key of the owner's account
+   * @returns {WalletInitializationBuilder} This wallet initialization builder
+   */
+  owner(address: string): this {
+    if (this._owners.length >= DEFAULT_M) {
+      throw new BuildTransactionError('A maximum of ' + DEFAULT_M + ' owners can be set for a multisig wallet');
+    }
+    if (!isValidPublicKey(address)) {
+      throw new BuildTransactionError('Invalid address: ' + address);
+    }
+    if (this._owners.includes(address)) {
+      throw new BuildTransactionError('Repeated owner address: ' + address);
+    }
+    this._owners.push(address);
+    return this;
+  }
+
+  protected initBuilder(tx: Transaction) {
+    super.initBuilder(tx);
+    const createAcc = tx.txBody.cryptoCreateAccount;
+    if (createAcc && createAcc.key && createAcc.key.thresholdKey) {
+      this.initOwners(createAcc.key.thresholdKey as proto.ThresholdKey);
+    }
+  }
+
+  private initOwners(keys: proto.ThresholdKey) {
+    if (keys.keys && keys.keys.keys) {
+      keys.keys.keys.forEach(key => {
+        this.owner(hex.encode(key.ed25519!));
+      });
+    }
+  }
+
+  /** @inheritdoc */
+  protected async buildImplementation(): Promise<Transaction> {
+    this._txBodyData.key = { thresholdKey: this.buildOwnersKeys() };
+    this._txBodyData.initialBalance = 0;
+    this._txBody.transactionFee = +new BigNumber(this._fee.fee); // validate
```

Comment:
What is the `= +` for?

Reply:
that makes it positive

Reply:
https://en.wikibooks.org/wiki/JavaScript/Operators - although not super sure unary operators work on BigNumber - Fabian can you confirm that? There might be like an absolute value or magnitude function helper I would prefer to call here

Reply:
In this case has the same effect as calling toNumber() funtion on the BigNumber object

`/**
   * Returns the value of this BigNumber as a JavaScript primitive number.
   *
   * Using the unary plus operator gives the same result.
*/
  toNumber(): number;`

Reply:
can we use abs instead? I'd rather use an explicit feature rather than implicit casting